torch/csrc/utils/python_arg_parser.h

```diff
@@ -153,6 +153,7 @@ struct PythonArgs {
   template<int N>
   inline std::array<at::Tensor, N> tensorlist_n(int i);
   inline std::vector<int64_t> intlist(int i);
+  inline c10::OptionalArray<int64_t> intlistOptional(int i);
```

Comment:
this looks fine, although none of these things use c10/IValue types so I'm not sure why we wold change that here.

Reply:
`OptionalArray` isn't really an IValue type.  It's just a vector wrapper that allows implicit conversion to ArrayRef.  I could move it to ATen if you want.